### PR TITLE
PP-8540 log the remote IP address when authorising a payment

### PR DESCRIPTION
**Pull request by gidsg**
## WHAT YOU DID
Log IP address in authorisation request log message and structured logging

## How to test
Tests should make sense

Application should log IP address in log message and structured logging when authorising a payment

**Question** Is it preferable to return a `null` value in structured logging if no IP present? Currently, the key value is not included in this case.

### Logging

- [ ] only emit log lines at ERROR level which require immediate attention from a support engineer. These will trigger a zendesk alert.

### Documentation

- [ ] Updated README.md for any of the following ?

* Introduced any new environment variables / removed existing environment variable
* Added new API / updated existing API definition

### CODE REVIEW

**Comment by gidsg** [comment]
OK @alexbishop1 as discussed off-of-Github, I've removed the Optional field but used it when serialising. If we're happy with this approach I can squash the commits!

File: src/main/java/uk/gov/pay/connector/gateway/util/AuthorisationRequestSummaryStringifier.java [hunk 1]
```diff
@@ -1,7 +1,9 @@
 package uk.gov.pay.connector.gateway.util;
 
+import org.apache.commons.cli.Option;
```
Comment (alexbishop1) [inline src/main/java/uk/gov/pay/connector/gateway/util/AuthorisationRequestSummaryStringifier.java]: Suspicious import 🧐

**Review by alexbishop1 — CHANGES_REQUESTED** [review]
I think it’s fine to not include anything if there isn’t an IP address (which I don’t think will happen in practice). Please remove the rogue import and squash the commits.

File: src/main/java/uk/gov/pay/connector/gateway/util/AuthorisationRequestSummaryStringifier.java (hunk 1 above)
Comment (gidsg) [reply src/main/java/uk/gov/pay/connector/gateway/util/AuthorisationRequestSummaryStringifier.java]: @alexbishop1 Good spot 👀  have removed and squashed in 36c3c60

**Review by alexbishop1 — APPROVED** [review]

**Pull request merged**